### PR TITLE
refactor displayBlock.go

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**
create transactionCounter struct, and remove global parameters from displayBlock.go

### CODE REVIEW

File: process/block/displayBlock.go [hunk 1]
```diff
@@ -51,32 +63,34 @@ func DisplayLogInfo(
 	numShards uint32,
 	selfId uint32,
 	dataPool dataRetriever.PoolsHolder,
+	txCounter *transactionCounter,
```
Comment (iulianpascalau) [inline process/block/displayBlock.go]: this func could have been made part of the transactionCounter structure. And all methods could have been made unexported.

File: process/block/displayBlock.go [hunk 2]
```diff
@@ -106,10 +120,10 @@ func createDisplayableShardHeaderAndBlockBody(
 	return tableHeader, shardLines
 }
 
-func displayTxBlockBody(lines []*display.LineData, body block.Body) []*display.LineData {
-	txCounterMutex.Lock()
-	txsCurrentBlockProcessed = 0
-	txCounterMutex.Unlock()
+func displayTxBlockBody(lines []*display.LineData, body block.Body, txCounter *transactionCounter) []*display.LineData {
```
Comment (iulianpascalau) [inline process/block/displayBlock.go]: this func could have been made part of the transactionCounter structure.

File: process/block/shardblock.go [hunk 3]
```diff
@@ -1398,3 +1408,9 @@ func (sp *shardProcessor) DecodeBlockHeader(dta []byte) data.HeaderHandler {
 
 	return &header
 }
+
+// TransactionCounter returns the object that keeps track of the total nr. of processed txs,
+// and txs processed in this block
+func (sp *shardProcessor) TransactionCounter() *transactionCounter {
```
Comment (iulianpascalau) [inline process/block/shardblock.go]: Might remove this, we do not want to be able to use it (yet)

File: process/block/displayBlock.go (hunk 1 above)
Comment (BeniaminDrasovean) [reply process/block/displayBlock.go]: Changed

File: process/block/displayBlock.go (hunk 2 above)
Comment (BeniaminDrasovean) [reply process/block/displayBlock.go]: done

File: process/block/shardblock.go (hunk 3 above)
Comment (BeniaminDrasovean) [reply process/block/shardblock.go]: done.

Comment (raduchis) [reply process/block/shardblock.go]: Not done yet. It is used only in the tests.

File: process/block/displayBlock.go [hunk 4]
```diff
@@ -106,10 +120,10 @@ func createDisplayableShardHeaderAndBlockBody(
...
+	txCounter.mutex.Lock()
+	txCounter.currentBlockTxs = 0
```
Comment (raduchis) [inline process/block/displayBlock.go]: Use local variables for this. There is no need for the locks and unlocks in the for. One lock and one unlock after the for should be enough.

Comment (BeniaminDrasovean) [reply process/block/displayBlock.go]: Done.

File: process/block/shardblock.go (hunk 3 above)
Comment (BeniaminDrasovean) [reply process/block/shardblock.go]: Was done, but the changes were not pushed yet.

**Pull request merged**